### PR TITLE
PubSub for events

### PR DESCRIPTION
**Pull request by shazow**
Related to #1.

### CODE REVIEW

File: publisher.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	Subscribe(chan<- Event)
+
+	// Unsubscribe stops the channel from receiving further events. Returns false if channel was not subscribed to start with.
+	// TODO: Unsubcribe(chan<- Event) bool
+
+	// Publish emits the Event to all the subscribers.
+	Publish(Event)
+}
+
+// NewPublisher creates a Publisher which blocks on all operations.
```
Comment (jzila) [inline publisher.go:67]: Stale name in comment: `s/NewPublisher/SyncPublisher`

File: publisher.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+func SyncPublisher() Publisher {
+	return &publisher{
+		subscribers: []chan<- Event{},
+	}
+}
+
+type publisher struct {
+	// TODO: Could make a lock-free version of this with a goroutine+broadcast channel. Not sure it would be worthwhile though.
```
Comment (jzila) [inline publisher.go:75]: I wouldn't bother. This is much more concise, even if it's technically less gothic.

Comment (shazow) [inline publisher.go:75]: Might do it as another swappable implementation. The other benefit is I'd have instantly-returning Publish(Event) without spawning a goroutine for every Publish.

File: publisher.go (hunk 1 above)
Comment (shazow) [inline publisher.go:67]: Good catch, thanks!

**Pull request merged**